### PR TITLE
[WIP] Issue578: Change nan value on spec file.

### PR DESCRIPTION
**Pull request by rhomspuron**
Implement the possibility to change the nan value for another value defined by the user
on spock. It uses the environment variable SpecNanValue.

### CODE REVIEW

**Comment by cpascual** [comment]
Just a suggestion:
Maybe it would make sense to allow setting this via `sardanacustomsettings.py` as well

File: src/sardana/macroserver/recorders/storage.py [hunk 1]
```diff
@@ -279,6 +279,16 @@ def __init__(self, filename=None, macro=None, **pars):
             self.setFileName(filename)
         self._expectedlabels = pars.get('labels', None)
 
+        # Configure nan value
+        self.nan_value = float('nan')
+
+        if macro is None:
```
Comment (cpascual) [inline src/sardana/macroserver/recorders/storage.py]: You probably mean "is not None"

**Comment by cpascual** [comment]
Given that the RecordList object now supports interpolation/extrapolation, is this feature still needed?

**Comment by rhomspuron** [comment]
We should implement the same solution for 1D channels. In addition if we activate the interpolation/extrapolation this solution will not work. 

The users prefer to have interpolation/extrapolation for some channels and not for all of them.

File: src/sardana/macroserver/recorders/storage.py (hunk 1 above)
Comment (rhomspuron) [reply src/sardana/macroserver/recorders/storage.py]: Yes, your right

**Comment by rhomspuron** [comment]
Close by mistake. I removed my remote branch

**Comment by reszelaz** [comment]
I just discussed this with @rhomspuron and below are our conlcussions.

We investigated a little bit how this is handled in SPEC, and there is no such configuration parameter. One may however implement a *macro hardware* plugin (something similar to a controller plugin in sardana) to return a value reserved for such cases e.g. `-1`, -`999`, `0`. The same possiblity exists in Sardana as well.

For the case when a channel is not available e.g. a hardware is disconnected, an attribute is not readable, it is recommended to disable this channel in the measurement group or exclude it from the measurement group.

We think that a global parameter that would apply to all the channels may be not enough in the case when the user would like to specify a different value for each channel. So implementing a generic solution much beyond the changes proposed in this PR.

We should not forget that sardana also allows to plugin custom recorders, and easily one could inherit from the Sardana default SPEC recorder and catch the cases when `NaN` arrives and handle them differently, depending on the channel, etc.

Having all this in mind we propose to close this PR. If anyone has different opinion about this feel free to comment here and we will get back to this topic.

**Comment by dschick** [comment]
I justed asked at xrayutilities for handling Nan values during reading the spec files. I actually like to have nans. So I agree that one might write a custom recorder for changing that if needed.

**Pull request closed without merging**